src/modules/household/hooks/useHouseholdMenuActions.tsx

```diff
@@ -0,0 +1,250 @@
+  HouseholdFieldsFragment,
+  RelationshipToHoH,
+  useDeleteEnrollmentMutation,
+} from '@/types/gqlTypes';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+interface Args {
+  refetchHousehold: VoidFunction;
+  loading?: boolean;
+  currentDashboardEnrollmentId?: string;
+  currentMembers: HouseholdClientFieldsFragment[];
+  project: ManageHouseholdProject;
+  household: HouseholdFieldsFragment;
+  canEdit?: boolean;
+}
+export function useHouseholdMenuActions({
+  refetchHousehold,
+  loading,
+  currentDashboardEnrollmentId,
+  currentMembers,
+  project,
+  household,
+  canEdit,
+}: Args) {
+  const [deleteEnrollment, { loading: deleteLoading, error: deleteError }] =
+    useDeleteEnrollmentMutation({ onCompleted: () => refetchHousehold() });
+
+  const [splitInitiator, setSplitInitiator] =
+    useState<HouseholdClientFieldsFragment | null>(null);
+
+  const { confirmHohDialog, onChangeHoh } = useChangeHoh({ refetchHousehold });
+
+  const hasMultipleHohs = useMemo(
+    () =>
+      currentMembers.filter(
+        (hc) => hc.relationshipToHoH === RelationshipToHoH.SelfHeadOfHousehold
+      ).length > 1,
+    [currentMembers]
+  );
+
+  const { changeRelationshipDialog, openChangeRelationshipDialog } =
+    useChangeRelationshipToHoh();
+
+  const getRowSecondaryActionConfigs = useCallback(
+    (row: HouseholdClientFieldsFragment): CommonMenuItem[] => {
+      const isSoleHoh =
+        row.relationshipToHoH === RelationshipToHoH.SelfHeadOfHousehold &&
+        !hasMultipleHohs;
+
+      const clientId = row.client.id;
+      const enrollmentId = row.enrollment.id;
+
+      // Hold off adding Assessment items now, desired behavior is unclear (open individual intakes, switch Enrollment context, etc?)
+      // const assessmentItems = [
+      //   {
+      //     key: 'intake assessment',
+      //     title: 'Intake Assessment',
+      //     Icon: IntakeAssessmentIcon as any as SvgIconComponent,
+      //     to: generateSafePath(EnrollmentDashboardRoutes.INTAKE, {
+      //       clientId,
+      //       enrollmentId,
+      //     }),
+      //     ariaLabel: `Go to ${clientBriefName(row.client)}'s Intake Assessment`,
+      //   },
+      //   {
+      //     key: 'exit assessment',
+      //     title: 'Exit',
+      //     Icon: ExitAssessmentIcon,
+      //     to: generateSafePath(EnrollmentDashboardRoutes.EXIT, {
+      //       clientId,
+      //       enrollmentId,
+      //     }),
+      //     ariaLabel: `Go to ${clientBriefName(row.client)}'s Exit Assessment`,
+      //     disabled: !!row.enrollment.inProgress,
+      //   },
+      // ];
+
+      const manageHouseholdItems = canEdit
+        ? [
+            {
+              key: 'assign hoh',
+              title: 'Make Head of Household',
+              Icon: PersonIcon,
+              onClick: () => onChangeHoh(row),
+              ariaLabel: `Make ${clientBriefName(row.client)} the Head of Household`,
+              disabled: isSoleHoh,
```

Comment:
Should we add disabledReason for other disabled actions on HoH (make HoH and change relationship)?

![Screenshot 2025-02-21 at 9 26 49 AM](https://github.com/user-attachments/assets/0edc5d21-ac51-42fa-b17e-3786a81e57d9)

Reply:
This predates your PR but I'm noticing we should also apply the disabled text more consistently:
- either "HoH" or "Head of Household", but not both
- maybe: either provide the instruction "Change HoH first" on all, or none(?)

Reply:
Good question. I want to run this by Ash, I don't think they've seen the new disabled label style (where they're embedded instead of tooltips).

The tooltip limitation is the same limitation that caused me to not be able to add the section labels (there should be labels after each divider). It's the fact that with MUI Menu each child has to be a MenuItem, and they can't be wrapped in divs or keyboard nav gets super wonky.

Reply:
Adding to design session